syntax = "proto3";

package pb;

import "google/protobuf/any.proto";
import "google/protobuf/empty.proto";


service Portrait {

    rpc Reload(ReloadRequest) returns (CommonResponse) {};
    rpc Status(google.protobuf.Empty) returns (StatusResponse) {}; 
    rpc Stop(StopRquest) returns (CommonResponse) {};
    
    rpc GetPortrait(GetPortraitRequest) returns (PortraitResponse) {};
    rpc UpdatePortrait(UpdatePortraitRequest) returns (PortraitResponse) {};
    
}


message CommonResponse {
  int32 code = 1;
  string description  = 2;
}

message ReloadRequest {
  google.protobuf.Any dicts = 1; 
} 

message StopRquest {
  optional int32 signal = 1; 
} 

message StatusResponse {
  int32 code = 1;
  google.protobuf.Any status = 2; 
}


message GetPortraitRequest {
  string apiVersion = 1;
  string metadata = 2;
  string type = 3;
  string userId = 4;
}


message UpdatePortraitRequest {
  string apiVersion = 1;
  string metadata = 2;
  string type = 3;
  google.protobuf.Any dicts = 4;
}


message PortraitResponse {
  int32 code = 1;
  string description  = 2;
  google.protobuf.Any results = 3; 
}